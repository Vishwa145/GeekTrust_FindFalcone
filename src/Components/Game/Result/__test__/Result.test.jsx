import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter as Router } from "react-router-dom";
import Result from "../Result";
import {cleanup} from "@testing-library/react"
import renderer from "react-test-renderer"

const result1=null;
const result2=undefined;
const result3={planet_name:"Enchai" ,status:"success"};
const result4={status:false};

afterEach(cleanup);

it("1.render successfully", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Router><Result result={result1}/></Router>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("2.render successfully", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Router><Result result={result2}/></Router>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("3.render successfully", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Router><Result result={result3}/></Router>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("4.render successfully", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Router><Result result={result4}/></Router>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("Matches snapshot", ()=>{
    const tree = renderer.create(<Router><Result result={result1}/></Router>).toJSON();
    expect(tree).toMatchSnapshot("1");
});

it("Matches snapshot", ()=>{
    const tree = renderer.create(<Router><Result result={result1}/></Router>).toJSON();
    expect(tree).toMatchSnapshot("2");
});

it("Matches snapshot", ()=>{
    const tree = renderer.create(<Router><Result result={result1}/></Router>).toJSON();
    expect(tree).toMatchSnapshot("3");
});

it("Matches snapshot", ()=>{
    const tree = renderer.create(<Router><Result result={result1}/></Router>).toJSON();
    expect(tree).toMatchSnapshot("4");
});