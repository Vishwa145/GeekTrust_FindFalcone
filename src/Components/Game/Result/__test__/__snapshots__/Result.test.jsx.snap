// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot: 1 1`] = `
<div
  className="result"
>
  <div
    className="answer"
  >
    <h1>
      <strong>
        In search
      </strong>
    </h1>
    <button>
      <a
        href="/"
        onClick={[Function]}
      >
        close
      </a>
    </button>
  </div>
</div>
`;

exports[`Matches snapshot: 2 1`] = `
<div
  className="result"
>
  <div
    className="answer"
  >
    <h1>
      <strong>
        In search
      </strong>
    </h1>
    <button>
      <a
        href="/"
        onClick={[Function]}
      >
        close
      </a>
    </button>
  </div>
</div>
`;

exports[`Matches snapshot: 3 1`] = `
<div
  className="result"
>
  <div
    className="answer"
  >
    <h1>
      <strong>
        In search
      </strong>
    </h1>
    <button>
      <a
        href="/"
        onClick={[Function]}
      >
        close
      </a>
    </button>
  </div>
</div>
`;

exports[`Matches snapshot: 4 1`] = `
<div
  className="result"
>
  <div
    className="answer"
  >
    <h1>
      <strong>
        In search
      </strong>
    </h1>
    <button>
      <a
        href="/"
        onClick={[Function]}
      >
        close
      </a>
    </button>
  </div>
</div>
`;
